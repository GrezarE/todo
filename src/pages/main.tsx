import React, { useEffect } from 'react';
import { TodoAdd } from '../components/todo-add/todo-add';
import { TodoList } from '../components/todo-list/todo-list';
import { useLocalStorageHook } from '../hooks/localHook';
import { useDispatch, useSelector } from '../redux/hooks';
import { getTasksFromLocal } from '../redux/todoSlice';

export const Main = () => {
  const { user, todoList } = useSelector((store) => store.todo);
  const [localTodos, setLocalTodos] = useLocalStorageHook(user, []);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getTasksFromLocal(localTodos));
  }, []);

  useEffect(() => {
    setLocalTodos(todoList);
  }, [todoList]);

  return (
    <div className='main' data-testid="main">
      <h1>Todo List</h1>
      <TodoAdd />
      <TodoList />
    </div>
  );
};
