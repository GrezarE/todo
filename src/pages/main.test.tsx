import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import { Main } from './main';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

describe('Main', () => {
  it('main', () => {
    const store = mockStore({
      todo: {
        user: 'user',
        todoList: []
      }
    });

    const main = render(
      <Provider store={store}>
        <Main />
      </Provider>
    );

    const box = screen.getByTestId('main');
    expect(main).toMatchSnapshot();
    expect(box).toBeInTheDocument();
  });
});
