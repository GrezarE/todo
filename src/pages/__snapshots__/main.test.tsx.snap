// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main main 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
        data-testid="main"
      >
        <h1>
          Todo List
        </h1>
        <form
          class="form"
        >
          <input
            placeholder="Добавить дело"
          />
          <button
            type="submit"
          >
            Добавить
          </button>
        </form>
        <ul
          class="grid"
        >
          <li
            class="header"
          >
            <p>
              №
            </p>
            <p>
              Задача
            </p>
            <p>
              Дата
            </p>
            <p>
              Статус
            </p>
            <p>
              Действия
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
      data-testid="main"
    >
      <h1>
        Todo List
      </h1>
      <form
        class="form"
      >
        <input
          placeholder="Добавить дело"
        />
        <button
          type="submit"
        >
          Добавить
        </button>
      </form>
      <ul
        class="grid"
      >
        <li
          class="header"
        >
          <p>
            №
          </p>
          <p>
            Задача
          </p>
          <p>
            Дата
          </p>
          <p>
            Статус
          </p>
          <p>
            Действия
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
